core/sdk-core/src/main/java/software/amazon/awssdk/core/compression/Compressor.java

```diff
@@ -33,15 +34,25 @@ public interface Compressor {
      */
     String compressorType();
 
+    /**
+     * Compress a byte array.
+     */
+    byte[] compress(byte[] content);
+
     /**
      * Compress an {@link InputStream}.
      */
     InputStream compress(InputStream inputStream);
 
     /**
-     * Compress an async stream.
+     * Compress a {@link ByteBuffer}.
+     */
+    ByteBuffer compress(ByteBuffer byteBuffer);
+
+    /**
+     * Compress an {@link AsyncRequestBody}.
      */
-    Publisher<ByteBuffer> compressAsyncStream(Publisher<ByteBuffer> publisher);
+    Publisher<ByteBuffer> compressAsyncRequestBody(Publisher<ByteBuffer> publisher);
 
     /**
      * Maps the {@link CompressionType} to its corresponding {@link Compressor}.
```

Comment:
Can we make CompressionType a pseudo-enum so we can define more values at runtime, possibly in different modules (prior art: `Region`)? We'd implement `forCompressorType` by doing `compressionType.newCompressor()`.

Reply:
refactored CompressionType, and moved forCompressorType() from Compressor to newCompressor() in CompressionType